ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/healthcheck/HealthCheckAgent.java

```diff
@@ -82,13 +94,47 @@ public String getName() {
     }
 
     @Override
-    public HealthCheckResponseDetail check(
-        final SimpleKsqlClient ksqlClient,
-        final URI serverEndpoint
-    ) {
+    public HealthCheckResponseDetail check(final HealthCheckAgent healthCheckAgent) {
       final RestResponse<KsqlEntityList> response =
-          ksqlClient.makeKsqlRequest(serverEndpoint, ksqlStatement);
+          healthCheckAgent.ksqlClient
+              .makeKsqlRequest(healthCheckAgent.serverEndpoint, ksqlStatement);
       return new HealthCheckResponseDetail(response.isSuccessful());
     }
   }
+
+  private static class KafkaBrokerCheck implements Check {
+    private final String name;
+
+    KafkaBrokerCheck(final String name) {
+      this.name = Objects.requireNonNull(name, "name");
+    }
+
+    @Override
+    public String getName() {
+      return name;
+    }
+
+    @SuppressFBWarnings("RV_RETURN_VALUE_IGNORED_NO_SIDE_EFFECT")
+    @Override
+    public HealthCheckResponseDetail check(final HealthCheckAgent healthCheckAgent) {
+      final String commandTopic = ReservedInternalTopics.commandTopic(healthCheckAgent.ksqlConfig);
+      boolean isHealthy;
+
+      try {
+        healthCheckAgent.serviceContext
+            .getAdminClient()
+            .describeTopics(Collections.singletonList(commandTopic))
+            .all()
+            .get();
```

Comment:
@rodesai @vcrfxia If Kafka is not available, this get() will wait for a couple of minutes. Do you think we should have a lower timeout? configurable timeout for healthchecks? or the default 2-3 mins is good?

Reply:
I think that we should use a shorter timeout here. The healthcheck runs every 30 seconds, so that seems appropriate as a timeout value.

Reply:
Done. I set a timeout to 30s